one_fm/api/tasks.py

```diff
@@ -517,20 +517,49 @@ def end_previous_shifts():
 		doc.submit()
 
 def create_shift_assignment(schedule, date):
-	shift_assignment = frappe.new_doc("Shift Assignment")
-	shift_assignment.start_date = date
-	shift_assignment.employee = schedule.employee
-	shift_assignment.employee_name = schedule.employee_name
-	shift_assignment.department = schedule.department
-	shift_assignment.post_type = schedule.post_type
-	shift_assignment.shift = schedule.shift
-	shift_assignment.site = schedule.site
-	shift_assignment.project = schedule.project
-	shift_assignment.shift_type = schedule.shift_type
-	shift_assignment.post_type = schedule.post_type
-	shift_assignment.post_abbrv = schedule.post_abbrv
-	shift_assignment.roster_type = schedule.roster_type
-	shift_assignment.submit()
+	try:
+		shift_assignment = frappe.new_doc("Shift Assignment")
+		shift_assignment.start_date = date
+		shift_assignment.employee = schedule.employee
+		shift_assignment.employee_name = schedule.employee_name
+		shift_assignment.department = schedule.department
+		shift_assignment.post_type = schedule.post_type
+		shift_assignment.shift = schedule.shift
+		shift_assignment.site = schedule.site
+		shift_assignment.project = schedule.project
+		shift_assignment.shift_type = schedule.shift_type
+		shift_assignment.post_type = schedule.post_type
+		shift_assignment.post_abbrv = schedule.post_abbrv
+		shift_assignment.roster_type = schedule.roster_type
+		shift_assignment.submit()
+	except Exception:
+			frappe.log_error(frappe.get_traceback())
+
+def overtime_shift_assignment():
+	"""
+	This method is to generate Shift Assignment for Employee Scheduling 
+	with roster type 'Over_Time'. It first looks up for Shift Assignment
+	of the employee for the day if he has any. Change the Status to "Inactive"
+	and proceeds with creating New shift Assignments with Roster Type OverTime.
+	"""
+	date = cstr(getdate())
+	now_time = now_datetime().strftime("%H:%M:00")
+	roster = frappe.get_all("Employee Schedule", {"date": date, "employee_availability": "Working" , "roster_type": "Over-Time"}, ["*"])
+	for schedule in roster:	

```

Comment:
Instead of iterating and then enqueuing the `process_overtime_shift` method, you can directly enqueue the method to process overtime shifts for all schedules by passing a list of schedules. This will reduce the number of background jobs created.
Modifications will have to be made to use a list instead of a schedule object in the `process_overtime_shift` method.